docs/technical-design/endpoints.md

```diff
@@ -0,0 +1,213 @@
+
+Also note that here we need the DataExportType that we just created in the domain model.
+
+#### Adding our method to the store
+
+Once we've defined this Prisma method, we can add it to our store, which will in turn make it available to our resolvers.
+_**Go to `/app-api/src/postgres/postgresStore.ts`**_.
+We'll import both the Prisma query we just made, as well as the domain-model type we created earlier.
+
+```typescript
+import { dataExport } from './dataExport'
+import { DataExportType } from '../domain-models/DataExport'
+```
+
+Next we'll add a new method both to the `Store` type, and the `NewPostgresStore`. First, in the type:
+
+```typescript
+dataExport: () => Promise<DataExportType | undefined>
+```
+
+And then in the NewPostgresStore:
+
+```typescript
+dataExport: () => dataExport(client),
+```
+
+When we access the store in our app, the dataExport method we defined above will be available to us.
+
+### Creating the resolver
+
+We use Apollo Client to interface with our data source and to provide information from the store to the front-end; the methods to do that are called "resolvers".  
+_**Create `/app-api/src/resolvers/dataExport.ts`**_.
+Here's the code.
+
+```typescript
+import { QueryResolvers } from '../gen/gqlServer'
+import { Store } from '../postgres'
+
+export function dataExportResolver(store: Store): QueryResolvers['dataExport'] {
+    return async (_parent, context) => {
+        const name = await store.dataExport()
+        if (name === undefined) {
+            return {
+                name: undefined,
+            }
+        }
+        return name
+    }
+}
+```
+
+The method we just put on the store gets called here. It will return the result of client.state.findFirst() that we defined above. The return type of `QueryResolvers['dataExport']` is a little mysterious. `QueryResolvers` is a generated file that will include the resolver we're defining here. We'll go over how to generate `QueryResolvers` a little further down.
+
+Now we have to add this resolver to our resolver configuration, so that it can be used on the front end.
+
+_**Go to `/app-api/src/resolvers/configureResolvers.ts`**_
+
+We'll import the resolver we just created.
+
+```typescript
+import { dataExportResolver } from './dataExport'
+```
+
+And also add it to the Resolvers object. We've made a query, so this line will be added to the query methods.
+
+```typescript
+dataExport: dataExportResolver(store),
+```
+
+### Generating code
+
+Once we have our backend code in place, we need to run a couple of commands to generate glue types and methods.
+
+_**On the command line, navigate to `/services/app-graphql`.**_  
+_**Run `yarn gqlgen`**_
+
+_**Again on the command line, navigate to `/services/app-api`**_  
```

Comment:
🤔 I *think* running `yarn prisma generate` should just happen first?  In the example you give, I don't think there is a db change, so you could prisma generate right away.  I mention this because once a developer runs the command they get the benefit of those new types while building the api part, which could be helpful to them.

Reply:
@haworku when you say "happen first"...where do you think the command should be run in the process described in the doc?

Reply:
I believe it is the first step to the graphql docs. 

If you make a change the prisma schema at all as part of your development of the api you would need to run `yarn prisma generate` to get the new client and types. If you don't make schema changes, you can skip the script all together. The dev tool is already running that command to run the api.  That's my understanding at least. @mojotalantikite please correct me if I am mistaken.

Reply:
There's actually a couple things that need to happen. `yarn prisma generate` will generate the schemas, but it won't make a new SQL command that alters the DB. 

If you need to add anything to the schema and alter the DB, you can do that and work with it locally using [`prisma db push`](https://www.prisma.io/docs/concepts/components/prisma-migrate/db-push). Once you know you have the change you want with the schema, you can do [`prisma migrate dev --name name_of_db_change`](https://www.prisma.io/docs/concepts/components/prisma-migrate/get-started). That will generate the SQL code that runs in the DB to change the tables properly for your new schema.